pkg/pool-manager/pool_test.go

```diff
@@ -64,20 +64,48 @@ var _ = Describe("Pool", func() {
 			table.Entry("Start: 02:00:00:00:00:00  End: 02:10:00:00:00:00", "02:00:00:00:00:00", "02:10:00:00:00:00", false),
 			table.Entry("Start: 02:FF:00:00:00:00  End: 02:00:00:00:00:00", "02:FF:00:00:00:00", "00:00:00:00:00:00", true),
 		)
+
+		table.DescribeTable("should check that the the first octet is 02, 06, 0A or 0E", func(MacAddr string, needToFail bool) {
+			MacAddrHW, err := net.ParseMAC(MacAddr)
+			Expect(err).ToNot(HaveOccurred())
+			err = checkCast(MacAddrHW)
+			if needToFail {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+		},
+			table.Entry("Valid address: 02:00:00:00:00:00", "02:00:00:00:00:00", false),
+			table.Entry("Valid address: 06:00:00:00:00:00", "06:00:00:00:00:00", false),
+			table.Entry("Valid address: 0A:00:00:00:00:00", "0A:00:00:00:00:00", false),
+			table.Entry("Valid address: 0E:00:00:00:00:00", "0E:00:00:00:00:00", false),
+			table.Entry("Invalid address: 01:FF:00:00:00:00, the first octet is not 02, 06, 0A or 0E", "01:00:00:00:00:00", true),
+			table.Entry("Invalid address: FF:FF:00:00:00:00, the first octet is not 02, 06, 0A or 0E", "FF:FF:00:00:00:00", true),
+		)
 	})
 
 	Describe("Pool Manager General Functions ", func() {
 		It("should create a pool manager", func() {
 			fakeClient := fake.NewSimpleClientset()
 			startPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
 			Expect(err).ToNot(HaveOccurred())
-			endPoolRangeEnv, err := net.ParseMAC("FD:FF:FF:FF:FF:FF")
+			endPoolRangeEnv, err := net.ParseMAC("02:FF:FF:FF:FF:FF")
 			Expect(err).ToNot(HaveOccurred())
 			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
 			Expect(err).ToNot(HaveOccurred())
 		})
 
 		It("should fail to pool manager because of the invalid range", func() {
+			fakeClient := fake.NewSimpleClientset()
+			startPoolRangeEnv, err := net.ParseMAC("0A:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			endPoolRangeEnv, err := net.ParseMAC("02:00:00:00:00:00")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = NewPoolManager(fakeClient, startPoolRangeEnv, endPoolRangeEnv, false)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("should fail to pool manager because of the first octet of RangeStart is not 02, 06, 0A, 0E", func() {
```

Comment:
should fail to create a pool manager object when the first octet of RangeStart is not 02, 06, 0A, 0E